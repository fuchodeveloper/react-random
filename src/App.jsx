import React, {Component} from 'react';
import axios from 'axios';
import alertify from 'alertify.js';

class App extends Component {
  state = {
    randomNumberState: [],
    selected: '',
  }

  handleChange = event => {
    this.setState({
      selected: event.target.value
    },
      () => (this.sortNumbers(this.state)));
  }

  sortNumbers = state => {
    let result;
    const { selected, randomNumberState } = state;

    if (selected === 'ascending') {
      result = randomNumberState.sort((a, b) => a - b);
    } else {
      result = randomNumberState.sort((a, b) => b - a);
    }

    this.setState(() => ({ randomNumberState: result }));
  }

  savePhonenumbers = () => {
    this.saveAction(this.state.randomNumberState);
  }

  saveAction = phoneNumbers => {
    return axios.post('/save', phoneNumbers)
      .then(function (response) {
        alertify.delay(900);
        alertify.logPosition('top right');
        alertify.success(response.data.message);
      })
      .catch(function (error) {
          alertify.delay(900);
          alertify.logPosition('top right');
          alertify.error(error.response.data.message);
      });
  }

  generatePhoneNumbers = () => {
    let randomNumberArray = [];

    const maxAllowed = 100;

    let array = new Uint32Array(maxAllowed);
    window.crypto.getRandomValues(array);

    for (let i = 0; i < array.length; i++) {
      const sliceNumber = array[i].toString().slice(0, 7);
      randomNumberArray.push(`080${sliceNumber}`);
    }
    this.setState(() => ({
      randomNumberState: randomNumberArray,
      selected: '',
    }));
  }

  render() {
    const { randomNumberState, selected } = this.state;
    const maxNumber = Math.max(...randomNumberState);
    const minNumber = Math.min(...randomNumberState);

    return (
      <div className="wrapper">
        <div className="center">
          <h1>Random phone number generator</h1>
          <p>Generate phone numbers: </p> 
          <button className="generate-button" onClick={this.generatePhoneNumbers}>Generate</button>

          {
            randomNumberState.length > 0 &&
            <div>
              <p>
                Total numbers generated: {randomNumberState.length}
              <span className="pipe">|</span>
                Max number: {`0${maxNumber}`}
              <span className="pipe">|</span>
                Min number: {`0${minNumber}`}
              <span className="pipe">|</span>
                Sort:
              <span className="pipe" />
                <select id="select" value={selected} onChange={this.handleChange}>
                  <option value="" disabled>-- select --</option>
                  <option value="ascending">Ascending</option>
                  <option value="descending">Descending</option>
                </select>
              </p>
              <div>
                <div className="numbers-container">
                  <h2>Phone Numbers</h2>
                  <div className="numbers-display">
                    {randomNumberState.map(number => <p key={number}>{number}</p>)}
                  </div>
                  <button 
                    className="save-button"
                    onClick={this.savePhonenumbers}
                  >Save</button>
                </div>
              </div>
            </div>
          }
          
        </div>
        
      </div>
    );
  }
}

export default App;
