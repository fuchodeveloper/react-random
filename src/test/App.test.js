import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { configure, shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

describe('<App />', () => {
  const wrapper = shallow(<App />);
  console.log('wrapper', wrapper.debug())

  // it('should contain H2 element', () => {
  //   expect(wrapper.find('h2').length).toBe(1);
  // })
});
