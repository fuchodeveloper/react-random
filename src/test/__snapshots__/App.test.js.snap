// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="center"
      >
        <h1>
          Random phone number generator
        </h1>
        <p>
          Generate phone numbers: 
        </p>
        <button
          className="generate-button"
          onClick={[Function]}
        >
          Generate
        </button>
      </div>,
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Random phone number generator
          </h1>,
          <p>
            Generate phone numbers: 
          </p>,
          <button
            className="generate-button"
            onClick={[Function]}
          >
            Generate
          </button>,
          false,
        ],
        "className": "center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Random phone number generator",
          },
          "ref": null,
          "rendered": "Random phone number generator",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Generate phone numbers: ",
          },
          "ref": null,
          "rendered": "Generate phone numbers: ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Generate",
            "className": "generate-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Generate",
          "type": "button",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="center"
        >
          <h1>
            Random phone number generator
          </h1>
          <p>
            Generate phone numbers: 
          </p>
          <button
            className="generate-button"
            onClick={[Function]}
          >
            Generate
          </button>
        </div>,
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Random phone number generator
            </h1>,
            <p>
              Generate phone numbers: 
            </p>,
            <button
              className="generate-button"
              onClick={[Function]}
            >
              Generate
            </button>,
            false,
          ],
          "className": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Random phone number generator",
            },
            "ref": null,
            "rendered": "Random phone number generator",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Generate phone numbers: ",
            },
            "ref": null,
            "rendered": "Generate phone numbers: ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Generate",
              "className": "generate-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Generate",
            "type": "button",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
